sdk-api-kotlin/src/main/kotlin/dev/restate/sdk/kotlin/api.kt

```diff
@@ -453,11 +492,31 @@ suspend inline fun <reified T : Any> AwakeableHandle.resolve(payload: T) {
   return this.resolve(KtSerdes.json(), payload)
 }
 
+/**
+ * A [DurablePromise] is a durable, distributed version of a Kotlin's Deferred, or more commonly of
+ * a future/promise. Restate keeps track of the [DurablePromise] across restarts/failures.
+ *
+ * You can use this feature to implement interaction between different workflow handlers, e.g. to
+ * send a signal from a shared handler to the workflow handler.
+ *
+ * Use [SharedWorkflowContext.promiseHandle] to complete a durable promise, either by
+ * [DurablePromiseHandle.resolve] or [DurablePromiseHandle.reject].
+ *
+ * A [DurablePromise] is tied to a single workflow execution and can only be resolved or rejected
+ * while the workflow run is still ongoing. Once the workflow is cleaned up, all its associated
+ * promises with their completions will be cleaned up as well.
+ *
+ * NOTE: This interface MUST NOT be accessed concurrently since it can lead to different orderings
+ * of user actions, corrupting the execution of the invocation.
+ */
```

Comment:
should we move a bit of this explanation to the ctx.promise JavaDocs? Because those are the ones you see first before clicking on the return type.

Reply:
This specific bit about not concurrent access is fine to keep it here.